tests/test.js

```diff
@@ -62,3 +75,57 @@ describe("Embroider build", function () {
     });
   }
 });
+
+describe("Build error on missing integrity hash for external resources", function () {
+  let buildOutput;
+
+  beforeEach(async function () {
+    this.timeout(60000);
+    try {
+      await execa("pnpm", ["build"], {
+        cwd: path.join(
+          __dirname,
+          "..",
+          "test-apps",
+          "external-resource-missing-integrity-hash",
+        ),
+      });
+    } catch (error) {
+      buildOutput = error.message;
+    }
+  });
+
+  it("Throws an error if integrity hash is missing", function () {
+    expect(buildOutput).to.include(
+      "🚨🚨 The following external resources do not have an integrity hash:\n" +
+        '\n<script src="https://ajax.googleapis.com/ajax/libs/jquery/3.7.1/jquery.min.js"></script>\n',
+    );
+  });
+});
+
+describe("No build error when integrity hash present for external resources", function () {
+  let buildOutput;
+
+  beforeEach(async function () {
+    this.timeout(60000);
+    try {
+      const output = await execa("pnpm", ["build"], {
+        cwd: path.join(
+          __dirname,
+          "..",
+          "test-apps",
+          "external-resource-with-integrity-hash",
+        ),
+      });
+      buildOutput = output.stdout;
+    } catch (error) {
+      buildOutput = error.message;
+    }
+  });
+
+  it("Does not throw an error if integrity hash is present", function () {
+    expect(buildOutput).to.not.include(
+      "🚨🚨 The following external resources do not have an integrity hash:\n",
+    );
+  });
+});
```

Comment:
This test is a little bit brittle. Such assertions are easy to get meaningless in a refactoring. E.g. it is easy to change the error message slightly and forgetting to adjust this test case as well. We can reduce that risk by asserting against the same string (variable) in both tests.